Don't consider it an error if testing fails because bmake is not executable.
This would happen if compiling for an architecture we can't run.
--- boot-strap.orig	2024-03-10 12:51:10.000000000 -0500
+++ boot-strap	2024-04-26 02:29:26.000000000 -0500
@@ -446,7 +446,13 @@
 	test) ;;
 	*) is_newer bmake tested || return;;
 	esac
-	Bmake test TEST_MAKE=$objdir/bmake "$@" || exit 1
+	Bmake test TEST_MAKE=$objdir/bmake "$@"
+	result="$?"
+	if [ "$result" -eq 126 ]; then
+		echo "WARNING: skipping tests because bmake is not executable, presumably because of cross compilation"
+	elif [ "$result" -ne 0 ]; then
+		exit 1
+	fi
 	touch tested
 }
 
